OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -64,10 +66,14 @@ public PathFinder(World world)
 		public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignoreActor)
 		{
 			var li = self.Info.TraitInfo<MobileInfo>().LocomotorInfo;
+			if (!cached)
+			{
+				domainIndex = world.WorldActor.Trait<DomainIndex>();
```

Comment:
I believe this should be TraitOrDefault and the null checks below left in. Otherwise you are making DomainIndex mandatory instead of just improving performance.